yt_dlp/extractor/common.py

```diff
@@ -2191,14 +2191,12 @@ def build_stream_name():
                     # contains EXT-X-STREAM-INF tag which references AUDIO
                     # rendition group but does not have CODECS and despite
                     # referencing an audio group it represents a complete
-                    # (with audio and video) format. So, for such cases we will
-                    # ignore references to rendition groups and treat them
-                    # as complete formats.
-                    if audio_group_id and codecs and f.get('vcodec') != 'none':
+                    # (with audio and video) format. But we can't know that
+                    # at this point so assume it's video only.
+                    # It can be resolved at later stage.
```

Comment:
How?

Reply:
Currently without this MR you would have just video file (either with audio or without). With this MR you will have audio file and video file (either with audio or without). When attempting to merge audio file with video file you could check if video file has audio already do respective steps.

Eg.
```python
if video_file.has_audio && video_file.audio_quality >= audio_file.quality:
  delete audio_file
  # Just keep original video file and ignore separate audio (current behavior)
else:
  video_file.remove_audio()
  video_file.merge(audio_file)
```